src/groupeddataframe/grouping.jl

```diff
@@ -1106,3 +1106,11 @@ function _aggregate(d::AbstractDataFrame, fs::AbstractVector,
     sort && sort!(res, headers)
     res
 end
+
+function DataFrames.DataFrame(gd::GroupedDataFrame)
+    length(gd) == 0 && return similar(parent(gd), 0)
+    # below we assume that gd.ends[end] == length(gd.idxs)
+    # and that gd.starts and gd.ends are increasing and cover a continuous range
+    gd.starts[1] == 1 && return parent(gd)[gd.idx, :]
```

Comment:
Isn't it always the case that `gd.starts[1]`? AFAICT that's what we explicitly set in `group_rows`.

OTC, if I'm wrong, is it really useful to distinguish this case? It should be quite cheap to create a view, right?

Reply:
OK - I will make a `view` - I wanted to avoid it in the most common case.
`gd.starts[1]` is greater than `1` if we have missing values in grouping and we set `skipmissing` to `true`.

Reply:
>  `gd.starts[1]` is greater than `1` if we have missing values in grouping and we set `skipmissing` to `true`.

Indeed. We should probably just do `starts .-= starts[2] .+ 1 and `ends .-= ends[2] .+ 1. We already do that for `groups`.

Reply:
But I understand you mean to do it in `groupby` related mechanics not here. Right? I think it would be OK, but then `gd.idx` should not include the leading indices that are pointing at `missing` so `length(gd.idx)` may be less that `nrow(parent(gd))` and the same for `gd.ends[end]` (but the relationsip `gd.ends[end] == length(gd.idxs)` would still hold) - I assume that we are not relying on the fact that `gd.idx` has the same number of elements as rows in the parent, but that you know this part of code better 😄.

If we took this approach then we could avoid making the view here and just index by `gd.idx`.

Reply:
Right. I don't think we assume `length(gd.idx) == nrow(parent(gd))`, but only test will tell for sure.

Reply:
OK, but I guess it should be a separate PR. Are willing to work on it?

Reply:
See https://github.com/JuliaData/DataFrames.jl/pull/1692. But actually when `sort=true` there's no guaranty that `gd.starts[1] == 1`, only that `minimum(gd.starts) == 1`. And using `gd.idx` directly would return groups in an unsorted order. AFAIK that's unavoidable since we only sort group indices (`gd.start` and `gd.stops`), which should be faster than sorting row indices (`gd.idx`) when there are few groups. Not sure whether sorting group indices would be significantly slower.

Reply:
OK - I have fixed it (that is why I have said that split-apply-combine is tricky 😄). We cannot avoid making a copy of `gd.idx` in this case unfortunately, but I have proposed the fastest method to get what we want I could think of.

The fix depends on #1692, so I will rebase this PR, when we merge #1692 (I could resize to make it independent, but I think it is better to follow the idea of code simplification that can be achieved with #1692).

Reply:
In parallel I was thinking about the same and consider using `view` instead of `copyto!` or creating a correct `rperm` from the start - it would probably help but would complicate the design a bit. But in general I am OK to drop #1692.

If you decide to drop #1692  I can fix this PR not to rely on #1692 and be safe.